packages/compose/src/hooks/use-merge-refs/index.js

```diff
@@ -36,19 +36,24 @@ export default function useMergeRefs( refs ) {
 		refs.forEach( ( ref, index ) => {
 			const previousRef = previousRefs.current[ index ];
 
-			if (
+			if ( ref && ref.hasOwnProperty( 'current' ) ) {
+				ref.current = element.current;
+			} else if (
 				typeof ref === 'function' &&
 				ref !== previousRef &&
 				didElementChange.current === false
 			) {
-				previousRef( null );
+				if ( previousRef ) {
+					previousRef( null );
+				}
+
 				ref( element.current );
 			}
 		} );
 
 		previousRefs.current = refs;
 		didElementChange.current = false;
-	}, refs );
+	}, [ refs.length, ...refs ] );
```

Comment:
Curious why this hook needed to be changed? was it a bug, should we add a unit test for it?

Reply:
It's only relevant if extra dependencies are added. Strangely, React doesn't trigger and update if you go from e.g. `[]` to `[ value ]`. It assumes the amount of dependencies is constant.

Reply:
Can add a test case though.